common/lib/client/realtimechannel.js

```diff
@@ -49,6 +49,53 @@ var RealtimeChannel = (function() {
 		return args;
 	};
 
+	RealtimeChannel.prototype.setOptions = function(options, callback) {
+		if(!callback) {
+			if (this.rest.options.promises) {
+				return Utils.promisify(this, 'setOptions', arguments);
+			}
+
+			callback = function(err){
+				if(err) {
+					Logger.logAction(Logger.LOG_ERROR, 'RealtimeChannel.setOptions()', 'Set options failed: ' + err.toString());
+				}
+			};
+		}
+		var err = validateChannelOptions(options);
+		if(err) {
+			callback(err);
+			return;
+		}
+		Channel.prototype.setOptions.call(this, options);
+		if(this._shouldReattachToSetOptions(options)) {
+			this._attach(true, callback);
+		} else {
+			callback();
+		}
+	};
+
+	function validateChannelOptions(options) {
+		if(options && 'params' in options && !Utils.isObject(options.params)) {
+			return new ErrorInfo('options.params must be an object', 40000, 400);
+		}
+		if(options && 'modes' in options){
+			if(!Utils.isArray(options.modes)){
+				return new ErrorInfo('options.modes must be an array', 40000, 400);
+			}
+			for(var i = 0; i < options.modes.length; i++){
+				var currentMode = options.modes[i];
+				if(!currentMode || typeof currentMode !== 'string' || !Utils.arrIn(Utils.modes, String.prototype.toUpperCase.call(currentMode))){
+					return new ErrorInfo('Invalid channel mode: ' + currentMode, 40000, 400);
+				}
+			}
+		}
+	}
+
+	RealtimeChannel.prototype._shouldReattachToSetOptions = function(options) {
+		// TODO: Check if the new options are different than the old ones
```

Comment:
Does this need to be addressed?

Reply:
It points to a possible performance optimization but we do not have a performance issue with this now.